### DISCUSSION TITLE
cannot access js file to html

### DISCUSSION THREAD

**Discussion by hwangheechan** [community/community#145919] [Programming Help]
### Select Topic Area

Question

### Body

HTML file

<!DOCTYPE html>
<html lang="ko">

<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Hometown</title>
    <link rel="stylesheet" href="202302718_styles.css">
</head>

<body>
    <script>
        window.alert("Hello! Welcome to my Hometown introduction!"); //output 3
    </script>

    
    <header id="Header">
        <h1>My HomeTown<span class="note"> "PyeongTaek"</span></h1>
        <p id="demo1">평택은 경기도 남서부에 위치한 도시로, 대한민국의 주요 산업과 물류 중심지 중 하나이다.</p>
    </header>

    
    <aside class="vertical-nav">
        <nav>
            <a href="#History">역사와 배경</a>
            <a href="#Attractions">관광 명소</a>

            
            <div class="dropdown">
                <a href="#LocalFood" class="dropbtn">로컬 푸드</a>
                <div class="dropdown-content">
                    <a href="#Rice">쌀</a>
                    <a href="#Stew">부대찌개</a>
                    <a href="#Pear">배</a>
                </div>
            </div>

            <a href="#FunFacts">신비한 이야기</a>
        </nav>
    </aside>

    
    <main>
        <section>
            <h2>블로그 읽기 전 간단한 설문조사</h2>
            <form action="/submit" method="post" onsubmit="return validateForm()">
                <label for="name">사이트를 방문한 이유:</label>
                <input type="text" id="name" name="name" placeholder="방문 이유"><br>

                <label for="email">평택에 관해 궁금한 것:</label>
                <input type="email" id="email" name="email" placeholder="아무거나 괜찮습니다"><br>
            </form>
        </section>

        <hr>

        <section>
            <h2>얼마나 평택을 잘 알고있나요?!</h2>
            <p>평택에는 몇 명의 인구가 살고 있을까요?</p>
            <button type="button" onclick="displayAnswer()"> Check answer </button> 
            <br><br>
            <p id="arithmetic"></p>
            <p id="assignment"></p>
            <p id="comparison1"></p>
            <p id="comparison2"></p>
        </section>
        <hr>

        
        <script>
            var mydong = 1;
            var dong = (1 + 17) * 1; // (나의 동 + 나머지 동) * 지역 수(평택 1)
            document.getElementById("arithmetic").innerHTML = "평택시는 " + dong + "개의 동 수를 가지고 있다."

            var area = 443;
            var area_meter = 53;
            area_meter *= 0.01; // km를 m로 치환
            area += area_meter;
            document.getElementById("assignment").innerHTML = "평택시의 크기는 " + area + " km이다.";

            let text1 = "Seoul";
            let text2 = "PyeongTaek";
            let result1 = text1 > text2;
            let result2 = text1 == text2;
            document.getElementById("comparison1").innerHTML = "서울이 평택보다 인구 수가 많은가요?" + " " + result1;
            document.getElementById("comparison2").innerHTML = "서울과 평택은 같은 경기도에 있나요?" + " " + result2;
        </script>

        <section id="History">
            <h2>시대별로 알아보는 평택의 역사와 배경</h2>
            <article>
                <h3>삼국 ~ 조선 시대</h3>
                <p>평택은 삼국시대 때, 삼국이 서로 차지하려던 지역이다. 한강 유역과 서해연안 근처에 위치한 평택을 차지해야 나라의 힘을 키울 수 있었기에 삼국이 서로 평택을 차지하려고 하였다.
                </p>
            </article>
            <article>
                <h3>일제 강점기 시대</h3>
                <p>평택은 일제 강점기 당시, 산업화와 농업 개발이 이루어졌다. 평택은 쌀 생산지로 유명했으며, 해안선을 따라 소규모 항구들이 발전했다.</p>
            </article>
            <article>
                <h3>1960년대 이후</h3>
                <p>평택은 1960년대 이후, 군사적, 경제적 중심지로 성장했다. 주한미군 기지가 위치해 있으며, 현대의 평택은 산업과 물류의 허브로 중요하다.</p>
            </article>
        </section>

        <hr>

        <section id="Attractions">
            <h2>평택에서 가볼만한 곳</h2>
            <table>
                <tr>
                    <th>이름</th>
                    <th>평택호 관광단지</th>
                    <th>소사벌 레포츠타운</th>
                    <th>평택국제중앙시장</th>
                </tr>
                <tr>
                    <td>유형</td>
                    <td><img src="pyeongtaekho.jpg" alt="평택호 관광단지"></td>
                    <td><img src="sport.webp" alt="소사벌레포츠타운"></td>
                    <td><img src="sijang.jpg" alt="평택국제중앙시장"></td>
                </tr>
                <tr>
                    <td>설명</td>
                    <td>평택호는 인공호수로 산책로와 공원이 조성되어 있어 휴식과 여가를 즐기기 좋다.</td>
                    <td>다양한 스포츠 시설이 있는 복합 레저 공간으로, 다양한 운동을 즐길 수 있다.</td>
                    <td>다양한 먹거리와 특산품을 즐길 수 있는 전통 시장.</td>
                </tr>
            </table>
        </section>

        <hr>

        <section id="LocalFood">
            <h2>평택의 로컬 푸드</h2>
            <div class="local-food-container">
                <div class="food-item" id="Rice">
                    <img src="rice.jpg" alt="쌀">
                    <p>쌀 🍚</p>
                </div>
                <div class="food-item" id="Stew">
                    <img src="food.JPG" alt="부대찌개">
                    <p>부대찌개 🍲</p>
                </div>
                <div class="food-item" id="Pear">
                    <img src="pear.png" alt="배">
                    <p>배 🍐</p>
                </div>
            </div>

            <br>
            <p id="demo2"></p>
            <p id="demo3"></p>
        </section>

        <hr>

        <script>
            var x;
            const y = "No! 아니다!"; // java statements and variables
            x = "여기서 질문! 라면은 평택의 로컬 푸드일까?";

            document.getElementById("demo2").innerHTML = x;
            document.getElementById("demo3").innerHTML = y;
        </script>

        <hr>

        <section id="FunFacts">
            <h2>평택의 흥미로운 사실</h2>
            <ol>
                <li>&#x2764; 평택에 삼성이 들어오면서 평택의 인구는 "급증"하였다</li>
                <li>&#128148; 평택은 아파트 미분양 순위가 전국에서 "1등"이다</li>
                <li>&#10084; 평택항 근처에서 오래된 난파선이 발견되고는 했다</li>
                <li>&#10084; 평택농악은 "유네스코 인류무형문화유산"에 등재되었다</li>
            </ol>
            <br>

            <p id="pt"></p>
            <br>
            <h3>평택 동네 인구 수 랭크</h3>
            <p id="ch"></p>

            <script>
                about_pt = {
                    location: "경기도",
                    elem: 40,
                    middle: 20,
                    high: 15,
                    university: 3
                }

                document.getElementById("pt").innerHTML = "평택에는 " + about_pt.location + " 에 위치해있다. 평택에는 많은 수의 학교가 존재한다. 초등학교는 " +
                    about_pt.elem + " 개, 중학교는 " + about_pt.middle + " 개, 고등학교는 " + about_pt.high + " 개, 대학교는 " + about_pt.university + " 개 정도가 있다.";

                const dong_rank = new Map([
                    ["평택동", 1],
                    ["송탄동", 2],
                    ["세교동", 3],
                    ["고덕동", 4],
                    ["안중읍", 5]
                ]);

                let ch = "";
                dong_rank.forEach(function (value, key) {
                    ch += key + '  (' + value + "등)" + "<br>"
                })

                document.getElementById("ch").innerHTML = ch;
            </script>

        </section>

        <hr>

        <section> 
            <h2>평택과 관련된 나의 정보</h2>
            <p id="info"></p>
            <p id="array"></p>
        </section>

        <script>
            const myinfo = new Set(["22years old", "KimDongHyun", "Segyo-dong"]);

            let text = "";
            for (const x of myinfo) {
                text += x + "<br>";
            }
            document.getElementById("info").innerHTML = "!! About my information !!" + "<br>" + text;

            const school = ["Segyo-elementry", "Shinhan-middle", "PyeongTaek-high"];
            document.getElementById("array").innerHTML = "<br>" + "1) My school: " + school;
        </script>

        <hr>

        <section id="Contact">
            <h2>How to Contact</h2>
            <form action="/submit" method="post" onsubmit="return validateForm()">
                <label for="name">이름:</label>
                <input type="text" id="name" name="name" placeholder="이름을 입력하세요"><br>

                <label for="email">이메일:</label>
                <input type="email" id="email" name="email" placeholder="이메일을 입력하세요"><br>

                <label for="phone">전화번호:</label>
                <input type="tel" id="phone" name="phone" placeholder="전화번호를 입력하세요"><br>

                <label for="inquiry">문의 유형:</label>
                <select id="inquiry" name="inquiry">
                    <option value="general">일반 문의</option>
                    <option value="technical">기술 지원</option>
                    <option value="billing">기타 문의</option>
                </select><br>

                <label for="city">거주지 선택:</label>
                <input list="cities" id="city" name="city" placeholder="거주 지역을 입력하세요">
                <datalist id="cities">
                    <option value="서울">
                    <option value="부산">
                    <option value="인천">
                </datalist><br>

                <label for="appointment">예약 날짜:</label>
                <input type="date" id="appointment" name="appointment"><br>

                <label for="time">예약 시간:</label>
                <input type="time" id="time" name="time"><br>

                <label for="comments">메시지:</label>
                <textarea id="comments" name="comments" rows="4" cols="50" placeholder="메시지를 입력하세요"></textarea><br>

                <button type="submit">Submit</button>
                <button type="reset">Reset</button>
            </form>
        </section>

        <hr>

        <section id="Media">
            <h2>유튜브에서 다룬 평택의 모습</h2>
            <iframe src="https://www.youtube.com/embed/FvOUHFOFfGY?si=v9bqnGjw5IqPLaJN&autoplay=1&mute=1"></iframe>
            <p>이 영상은 평택의 송탄으로, 미군 부대 앞의 모습을 보여준다.</p>
        </section>

        <hr>

        <section> 
            <h2>평택을 잘 공부했는지 알아볼까요?</h2>
            <p id="logical1"></p><br>
            <p id="logical2"></p><br>

            <p>유네스코 인류무형문화유산에 등재된 것은 무엇일까?</p> 
            <button onclick="document.getElementById('click1').innerHTML= '평택농악'"> check answer</button>
            <p id="click1"></p><br>

            <p>유튜브 영상 속 장소는 어디일까?</p>
            <button onclick="document.getElementById('click2').innerHTML= '송탄'"> check answer</button>
            <p id="click2"></p><br>

            <h3>평점을 남겨주세요!</h3>
            <p>1-5까지의 평점을 남겨주세요! 좋은 평점은 저에게 도움이 됩니다!</p>
            <input id="num" type="text">
            <button type="button" onclick="rate()">Thank You</button>
            <p id="error"></p>

        </section>

        <script>
            let check_dong = 18; // logical operators
            let check_area = 442.12;

            let usarmy = 1;
            document.getElementById("logical1").innerHTML = "평택은 18개의 동과 442.12km의 면적을 가진다." + "<br>" + (check_dong == 18 && check_area == 443.53);
            document.getElementById("logical2").innerHTML = "평택에는 주한 미군기지가 있는가??" + "<br>" + !(usarmy == 0);

        </script>
    </main>

    
    <footer id="Footer">
        <p>Author: Hankuk University of Foreign Studies</p>
        <p>출처: 유튜브, 구글, 나무위키</p>
        <p>TEL. 010-8377-3412 | <a href="https://donghyuneee.github.io/">Github Blog</a></p>
        <p>&copy; Hankuk University of Foreign Studies. All Rights Reserved.</p>
        <p><a href="#Header">Page up</a></p>
    </footer>

    <script>
        document.getElementById("demo1").innerHTML = "평택은 경기도 남서부에 위치한 도시로, 대한민국의 주요 산업과 물류 중심지 중 하나이다." + " " + "평택이라는 지명은 평평한 평야라는 지역 특징에서 유래되었다."; // output 1
        window.alert("Contact me if you need help!"); //output 3
        console.log("written by Dong Hyun Kim"); //output 4
        console.log("written 2024-11-27"); //output 4

    </script>

    <script src="202302718_script.js"></script>
</body>

</html>

JS FIle

function validateForm() { // 블로그 읽기 전 간단한 설문조사와 How to contact 에서 두 번 재사용하였다!
    var name = document.getElementById("name").value;
    var email = document.getElementById("email").value;
    if (name == "" || email == "") {
        alert("이름과 이메일 필드는 필수입니다!");
        return false;
    }
    return true;
}

function displayAnswer() {
    document.getElementById("answer").innerHTML = "평택의 인구는 약 575,032명입니다.";
}

function rate() {
    const message = document.getElementById("error");
    message.innerHTML = ""; // At beginning it is empty message 
    let x = document.getElementById("num").value; // Take input from a user
    try {
        if (x.trim() == "") throw "올바르지 않은 값입니다";
        if (isNaN(x)) throw "숫자로 표현해주세요!";
        x = Number(x);
        if (x < 1) throw "너무 적은 평점입니다..";
        if (x > 5) throw "마음만 받겠습니다 ㅎㅎ";
    }
    catch (err) {
        message.innerHTML = "" + err; // At the end it is error message 
        console.error(err);
    }
}

**Comment by iremozdemr** [comment]
- **Use a Relative Path**

Ensure you are using the correct relative path for your JavaScript file. If the file is in the same folder as your HTML file, try adding a ./ prefix to the file path in your <script> tag:

`<script src="./202302718_script.js"></script>`
If the JavaScript file is in a subfolder, specify the folder name in the path. For example, if the file is located in a folder named js, use:

`<script src="js/202302718_script.js"></script>`
If the file is in a parent directory, use ../:

`<script src="../202302718_script.js"></script>`

- **Ensure File Order**

JavaScript files are executed in the order they appear in the HTML. If a script depends on another, make sure the dependent script is included after its dependency.

For example:

```
<script src="dependency.js"></script>
<script src="main.js"></script>

```
- **Place Scripts at the Right Location**

If your script interacts with elements on the page, place the <script> tag before the closing </body> tag. This ensures the DOM is fully loaded before the script executes.

```
<body>
    <!-- Page content -->
    <script src="./202302718_script.js"></script>
</body>
```
Alternatively, use the defer attribute to load the script without blocking HTML parsing. This works well for scripts located in the <head>:

`<script src="./202302718_script.js" defer></script>`

**Comment by BelakbirSaid** [comment]
Check the file path in <script src="./202302718_script.js" defer></script> @hwangheechan

**Comment by iufor** [comment]
I have same issue

**Comment by anirudha-8** [comment]
# Use relative and correct path for js file

- for same directory level --> `<script src="./202302718_script.js"></script>`

- for same directory level but in another folder --> `<script src="./folderName/202302718_script.js"></script>`

- for parent directory level --> `<script src="../202302718_script.js"></script>`

---

**Note:-** Please check whether JavaScript is allowed in your browser or not.

**Comment by hwangheechan** [comment]
thank you

**Reply by anirudha-8** [reply]
@hwangheechan hey, can you able to access the js file or not?

**Reply by hwangheechan** [reply]
@anirudha-8 I solved it!

**Reply by anirudha-8** [reply]
> @anirudha-8 I solved it!

I am glad, you did it
